### PR TITLE
Add Eslint to backend and set up precommit lint hook

### LINKED ISSUE #59: Add ESLint to backend
**Is your feature request related to a problem? Please describe.**
The backend code should be uniform and have good style guidelines.

**Describe the solution you'd like**
Setup ESLint rules and make husky fail the commit if any rules generate warnings.
You can use common sense to select what rules should be enabled. Make sure that you have added rules specifically for nest and for typescript.

**Describe alternatives you've considered**
We currently have prettier but it does not enforce style

### Checklist:

- [x] I have double checked that the issue does not currently exist. 
- [x] I have added the appropriate labels (e.g. feature, front-end, back-end).
- [x] I have added the **Platform Project** project

### PR DESCRIPTION
**Pull request by oscarli00**
# Description
Add eslint to the backend and make it so that commits fail if there are any eslint errors or warnings.

Fixes/resolves #59 

## Screenshots
Please include any screenshots or media that illustrates the changes made

## Type of change
Please delete options that are not relevant.

- [x] **New feature** (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:
Leave blank if not applicable

I have completed these steps when making this pull request:
- [x] I have assigned my name to the issue
- [x] I have moved the issue to the **In Progress** column
- [x] I have labelled the PR appropriately
- [x] I have assigned myself to the PR

Before opening the PR for review:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have moved the linked issue to the **Review in Progress** column

### CODE REVIEW

**Review by justcrossheaven — DISMISSED** [review]
LGTM

**Review by uoa-aron527 — DISMISSED** [review]
Yep LGTM :-)

File: packages/backend/.eslintrc.js [hunk 1]
```diff
@@ -20,6 +20,5 @@ module.exports = {
     '@typescript-eslint/interface-name-prefix': 'off',
     '@typescript-eslint/explicit-function-return-type': 'off',
```
Comment (j-chad) [inline packages/backend/.eslintrc.js:21]: could you please quickly just explain the motivations behind these settings

Comment (j-chad) [reply packages/backend/.eslintrc.js:21]: also `    '@typescript-eslint/explicit-module-boundary-types': 'off',` it got cut off

Comment (ljey973) [reply packages/backend/.eslintrc.js:21]: These settings were already there, we decided to leave them apart from the rule that allowed the type any. These rules just allow interfaces to be named without an 'I' and allow functions to exist without an explicit return type.

Comment (j-chad) [reply packages/backend/.eslintrc.js:21]: oh can we make functions have an explicit return please

Comment (j-chad) [reply packages/backend/.eslintrc.js:21]: I think it aids a lot in readability

Comment (j-chad) [reply packages/backend/.eslintrc.js:21]: or at least just `@typescript-eslint/explicit-module-boundary-types`

Comment (j-chad) [reply packages/backend/.eslintrc.js:21]: `@typescript-eslint/explicit-function-return-type` is probably fine off

Comment (ljey973) [reply packages/backend/.eslintrc.js:21]: '@typescript-eslint/explicit-module-boundary-types': 'off' does the same as explicit-function-return-type except for exported functions, would you like that to be enforced as well?

**Review by uoa-aron527 — APPROVED** [review]
The changes made LGTM :-)

**Review by justcrossheaven — APPROVED** [review]
LGTM

File: packages/backend/.eslintrc.js (hunk 1 above)
Comment (j-chad) [reply packages/backend/.eslintrc.js:21]: yes I would prefer `@typescript-eslint/explicit-module-boundary-types` to be set to `warn`. not so fussed about `@typescript-eslint/explicit-function-return-type`

**Pull request merged**